julia/1.julia-setup.md

```diff
@@ -13,19 +11,70 @@ For Windows users, please open execute the following command in a `cmd`,
 
-To open a Julia REPL, just type `julia` in a terminal or click the Julia Icon in Windows.
 
-##### For users suffering from the slow download speed
+#### For users suffering from the slow download speed
+            
+You may need to specify another server for installing Juliaup. To do so, execute the following command in your terminal before running the script above.
 
+**Linux and macOS**
+```bash
+export JULIAUP_SERVER=https://mirror.nju.edu.cn/julia-releases/
+```
+**Windows**
+```PowerShell
+$env:JULIAUP_SERVER="https://mirror.nju.edu.cn/julia-releases/"
+```
 An alternative approach is downloading the corresponding Julia binary from the [Nanjing university mirror website](https://mirror.nju.edu.cn/julia-releases/).
 After installing the binary, please set the Julia binary path properly if you want to start a Julia REPL from a terminal, check this [manual page](https://julialang.org/downloads/platform/) to learn more.
 
-#### Step 2. Configure the package source and `Revise` in the startup file
+
+#### Installing Julia
+Once you have installed Juliaup, you can install Julia by running the following command in your terminal.
+  ```bash
+  juliaup default 1.9
+  ```
+To verify that Julia is installed, run the following command in your terminal.
+  ```bash
+  julia
+  ```
+- It should start a Julia REPL(Read-Eval-Print-Loop) session like this
+![REPL Session](https://miro.medium.com/v2/resize:fit:4800/format:webp/1*lxjWRvH3EzSa1N3Pg4iNag.png)
+- If you wish to install a specific version of Julia, please refer to the [documentation](https://github.com/JuliaLang/juliaup).
+
+### Step 2: Package Management
+- `Julia` has a mature eco-system for scientific computing.
+- 'Pkg' is the built-in package manager for Julia.
+- To enter the package manager, press `]` in the REPL.
+![PackageMangement](https://github.com/exAClior/QMBCTutorial/blob/ys/julia-tutorial/notebooks/resources/scripts/Packages.gif?raw=true)
+- The environment is indicated by the `(@v1.9)`.
+- To add a package, type `add <package name>`.
+- To exit the package manager press `backspace` key
+
+#### First Package: PkgServerClient.jl
+- Due to some special reasons, we need to install `PkgServerClient.jl` first.
+- You should first execute the following command
+
+  **Linux and macOS**
+  ```bash
+    export JULIA_PKG_SERVER='https://mirrors.nju.edu.cn/julia'
+  ```
+
+  **Windows**
+  ```PowerShell
+    $env:JULIA_PKG_SERVER='https://mirrors.nju.edu.cn/julia'
+  ```
+- Then you can add `PkgServerClient.jl` by typing `add PkgServerClient` in the package manager.
+- To make sure Julia uses the package automatically at startup, configure the `startup.jl` file.
+
+#### More Packages
+- You may find more packages for scientific computing [here](https://juliahub.com/).
+
+### Step 3. Configure the startup file and add `Revise`
 First create a new file `.julia/config/startup.jl` with the following content
```

Comment:
Not sure new users can find this path.

Reply:
I added more description, I will make sure to have them ask on zulip if they can't find it. If I need to explain in a way where brand new people can find it, this tutorial might go off the topic.